environment.yml

```diff
@@ -0,0 +1,10 @@
+name: sentiment
+channels:
+    - default
```

Comment:
you know with pip you can pip install <name> or, in some cases like with tensorflow, you would pip install with a url instead? Well conda (the package manager that uses the `environment.yml` to create a venv) can do the same thing; it can install packages from outside sources as well. To do this, you have to tell conda, in the env file, what sources (aka channels) to look in. Since we are not using any outside sources, we dont even need to list `default` in the `channels` section but i did it to make things easier for development (dont have to google the format etc).

Reply:
hmm thats interesting, what are the trade offs of using conda vs pip?

Reply:
Conda > pip in every way:
  * comes with venv creation out of box
  * each venv can even have it's own version of python in it
  * identifies/resolves package dependency conflicts
  * works with pip i.e. if you have a package that needs to be pip installed, you can pip install it within the conda env and even specify packages that should be pip installed by conda, in the `environment.yml` ([example file](https://github.com/jkarimi91/snake/blob/master/environment.yml))
  * it is widely used within the community especially among data scientists, researchers, research/ml engineers and the like; both pytorch and tensorflow list conda install instructions on their install page

Conda is a no brainer. The only time you really see people use pip is when they are first starting out with python and programming. 

An additional note about `channels`, the only time you will really every see a `channel` other than default, is with `pytorch` since facebook hosts torch/pytorch on their own channel at the moment.